tools/wptrunner/wptrunner/testrunner.py

```diff
@@ -558,6 +569,16 @@ def start_test_runner(self):
         self.logger.debug("Test runner started")
         # Now we wait for either an init_succeeded event or an init_failed event
 
+
+    def get_executor_for_test_type(self, test_type):
```

Comment:
I don't like this being different to the way we handle setting the initial executor. I'm also nervous that it depends on the class instance being serializable cross-process. I think maybe `start_runner` and `set_executor` should take `(executor_cls, executor_kwargs, executor_browser_cls, executor_browser_kwargs)` (maybe wrapped in a named tuple) and the `TestRunner` class should have a `set_executor` method that's used from `__init__` and when passed the `set_executor` message that contructs the executor.

Reply:
Thanks, updated but not exactly same as what you suggested, because `set_executor` needs to do something additional, i.e. tear down previous executor and setup the new one.

PTAL